File: examples/2015-day-01/elixir/test/aoc_test.exs
defmodule AocTest do
  use ExUnit.Case
  doctest Aoc

  @part_one_examples [
    {"(())", 0},
    {"()()", 0},
    {"(((", 3},
    {"(()(()(", 3},
    {"))(((((", 3},
    {"())", -1},
    {"))(", -1},
    {")))", -3},
    {")())())", -3}
  ]

  @part_two_examples [
    {")", 1},
    {"()())", 5}
  ]

  test "solves part one" do
    for {input, expected} <- @part_one_examples do
      assert Aoc.solve_part_one(input) == expected,
             "failed for input #{inspect(input)}"
    end
  end

  test "solves part two" do
    for {input, expected} <- @part_two_examples do
      assert Aoc.solve_part_two(input) == expected,
             "failed for input #{inspect(input)}"
    end
  end
end

File: examples/2015-day-01/typescript/tests/aoc_test.ts
import { assertEquals } from "@std/assert";

import { solvePartOne, solvePartTwo } from "../src/aoc.ts";

const partOneCases: [string, number][] = [
  ["(())", 0],
  ["()()", 0],
  ["(((", 3],
  ["(()(()(", 3],
  ["))(((((", 3],
  ["())", -1],
  ["))(", -1],
  [")))", -3],
  [")())())", -3],
];

const partTwoCases: [string, number][] = [
  [")", 1],
  ["()())", 5],
];

Deno.test("part one solves the examples", () => {
  for (const [input, expected] of partOneCases) {
    assertEquals(solvePartOne(input), expected, `failed on input ${JSON.stringify(input)}`);
  }
});

Deno.test("part two solves the examples", () => {
  for (const [input, expected] of partTwoCases) {
    assertEquals(solvePartTwo(input), expected, `failed on input ${JSON.stringify(input)}`);
  }
});
